python/lsst/metaserv/schemaToMeta.py

```diff
@@ -0,0 +1,271 @@
+
+                                          # units
+                        if self._isUnitLine(line):
+                            in_col["unit"] = self._retrUnit(line)
+
+                                          # ucds
+                        if self._isUcdLine(line):
+                            in_col["ucd"] = self._retrUcd(line)
+
+        iF.close()
+        return table
+
+    ###########################################################################
+    # Helper functions
+    ###########################################################################
+
+    def _isIndexDefinition(self, c):
+        return c in ["PRIMARY", "KEY", "INDEX", "UNIQUE"]
+
+    def _isCommentLine(self, str):
+        return re.match('[\s]*--', str) is not None
+
+    def _isUnitLine(self, str):
+        return re.search(r'<unit>(.+)</unit>', str) is not None
+
+    def _isUcdLine(self, str):
+        return re.search(r'<ucd>(.+)</ucd>', str) is not None
+
+    def _retrUnit(self, str):
+        xx = re.compile(r'<unit>(.+)</unit>')
+        x = xx.search(str)
+        return x.group(1)
+
+    def _retrUcd(self, str):
+        xx = re.compile(r'<ucd>(.+)</ucd>')
+        x = xx.search(str)
+        return x.group(1)
+
+    def _containsDescrTagStart(self, str):
+        return re.search(r'<descr>', str) is not None
+
+    def _containsDescrTagEnd(self, str):
+        return re.search(r'</descr>', str) is not None
+
+    def _retrDescr(self, str):
+        xx = re.compile(r'<descr>(.+)</descr>')
+        x = xx.search(str)
+        return x.group(1)
+
+    def _retrDescrStart(self, str):
+        xx = re.compile('<descr>(.+)')
+        x = xx.search(str)
+        return x.group(1)
+
+    def _retrDescrMid(self, str):
+        xx = re.compile('[\s]*--(.+)')
+        x = xx.search(str)
+        return x.group(1)
+
+    def _retrDescrEnd(self, str):
+        if re.search('-- </descr>', str):
+            return ''
+        xx = re.compile('[\s]*--(.+)</descr>')
+        x = xx.search(str)
+        return x.group(1)
+
+    def _retrIsNotNull(self, str):
+        if re.search('NOT NULL', str):
+            return '1'
+        return '0'
+
+    def _retrType(self, str):
+        arr = str.split()
+        t = arr[1]
+        if t == "FLOAT(0)":
+            return "FLOAT"
+        return t
+
+    def _retrDefaultValue(self, str):
+        if re.search(' DEFAULT ', str) is None:
+            return None
+        arr = str.split()
+        returnNext = 0
+        for a in arr:
+            if returnNext:
+                return a.rstrip(',')
+            if a == 'DEFAULT':
+                returnNext = 1
+
+    # example strings:
+    # "    PRIMARY KEY (id),",
+    # "    KEY IDX_sId (sId ASC),",
+    # "    KEY IDX_d (decl DESC)",
+    # "    UNIQUE UQ_AmpMap_ampName(ampName)"
+    # "    UNIQUE UQ_x(xx DESC, yy),"
+
+    def _retrColumns(self, str):
+        xx = re.search('[\s\w_]+\(([\w ,]+)\)', str.rstrip())
```

Comment:
xx is a terrible variable name.  A comment about what the pattern is doing would help, too (or a docstring for the function).